### DISCUSSION TITLE
How to extract sub URL using regex.

### DISCUSSION THREAD

**Discussion by Davidlimbong** [teamalgoritma/community#386] [Q&A] [answered]
Hello Team, mau tanya terkait pattern regex untuk extract substring dari URL. 
misal saya punya url `https://github.com/teamalgoritma/community/discussions/` yang saya ingin ambil `github.com` saja. Mohon bantuannya ya.

**Comment by iqbalbasyar** [comment, marked as answer]
Try the following with img tag (insensitive, multiline, global)

`^(?:https?:\/\/)?(?:[^@\n]+@)?(?:www\.)?([^:\/\n?]+)`

**Reply by Davidlimbong** [reply]
It works!! just a note if anyone want to use this on R please use double escape `\\` rather than `\`

**Comment by onlyphantom** [comment]
From this [StackOverflow answer](https://stackoverflow.com/questions/22721653/regex-trimming-url-to-root), you can also try:
```
https?:\/\/(?:.*\.)*(.+\..+?)\/
```

- The `http` part is self-explanatory, the `s?` makes the `s` optional so it matches `http` or `https`
- The `\/\/` escapes the two slashes (`//`)
- Then `(?:.*\.)*` matches if there is a `.` for subdomain, so `https://api.github.com` would still return `github.com` as well as `https://github.com`
- `(.+\..+?)` matches everything after it and `\/` matches the trailing `/` in `www.github.com/community/`

**Reply by Davidlimbong** [reply]
Hey Thanks Sam, this one also works for me.
